### PR TITLE
Code base update

### PR DESCRIPTION
**Pull request by alain-andre**
Bump to ruby 2.5, add api_keys easier management.

This PR is the first, the second one is https://github.com/Mapotempo/router-wrapper/pull/11

### CODE REVIEW

File: api/v01/matrix.rb [hunk 1]
```diff
@@ -80,9 +80,10 @@ class Matrix < APIBase
           failure: Status.failure,
         }
         get do
+          count :matrix
           params[:speed_multiplier] = params[:speed_multiplicator] if !params[:speed_multiplier]
           params[:speed_multiplier_area] = params[:speed_multiplicator_area] if params[:speed_multiplicator_area] && params[:speed_multiplicator_area].size > 0 && (!params[:speed_multiplier_area] || params[:speed_multiplier_area].size == 0)
-          matrix params
+          matrix params, operation: :matrix
```
Comment (fab-girard) [inline api/v01/matrix.rb:84]: 2nd params useless

File: api/v01/matrix.rb [hunk 2]
```diff
@@ -118,6 +120,7 @@ def matrix(params)
 
           results = RouterWrapper::wrapper_matrix(APIBase.services(params[:api_key]), params)
           results[:router][:version] = 'draft'
+          count_incr :operation, transactions: APIBase.count_locations(params[:dst]) * APIBase.count_locations(params[:src])
```
Comment (fab-girard) [inline api/v01/matrix.rb]: `count_incr :matrix, ...`

File: api/v01/route.rb [hunk 3]
```diff
@@ -216,10 +218,12 @@ def compute_routes(params)
               end
             end
           }
+          count_incr :operation, transactions: APIBase.count_locations(params[:locs])
```
Comment (fab-girard) [inline api/v01/route.rb]: `count_incr :routes, ...`

File: api/v01/route.rb [hunk 4]
```diff
@@ -85,16 +85,16 @@ class Route < APIBase
...
           optional :precision, type: Integer, default: 6, desc: 'Precison for encoded polyline.'
           optional :with_summed_by_area, type: Boolean, default: false, desc: 'Returns way type detail when set to true.'
         }
         get do
+          count :routes
           params[:locs] = [params[:loc].each_slice(2).to_a]
           params[:speed_multiplier] = params[:speed_multiplicator] if !params[:speed_multiplier]
           params[:speed_multiplier_area] = params[:speed_multiplicator_area] if !params[:speed_multiplier_area]
-          present compute_routes(params)[0], with: RouteResult, geometry: params[:geometry], toll_costs: params[:toll_costs], with_summed_by_area: params[:with_summed_by_area]
+          present compute_routes(params, operation: :routes)[0], with: RouteResult, geometry: params[:geometry], toll_costs: params[:toll_costs], with_summed_by_area: params[:with_summed_by_area]
```
Comment (fab-girard) [inline api/v01/route.rb:97]: hash with `:operation` key useless

File: api/v01/route.rb [hunk 5]
```diff
@@ -216,10 +218,12 @@ def compute_routes(params)
               end
             end
           }
+          count_incr :operation, transactions: APIBase.count_locations(params[:locs])
+          routes
         end
 
-        def many_routes(params)
-          routes = compute_routes(params)
+        def many_routes(params, operation: nil)
```
Comment (fab-girard) [inline api/v01/route.rb:225]: 2nd parameter useless

File: test/api/v01/isoline_test.rb [hunk 6]
```diff
@@ -67,4 +67,13 @@ def test_isoline_limit_should_not_pass
     get '/0.1/isoline', api_key: 'demo_limit', src: '43.2804,5.3806,43.291576,5.355835', dst: '43.2804,5.3806,43.291577,5.355836'
     refute last_response.ok?, last_response.body
   end
+
+  def test_count_isolines
+    [:get, :post].each_with_index do |method, indx|
+      send method, '/0.1/isoline', {api_key: 'demo', loc: '43.2804,5.3806', size: 1}
+      keys = RouterWrapper.config[:redis_count].keys("router:isolines:#{Time.now.utc.to_s[0..9]}_key:demo_ip*")
+      assert_equal 1, keys.size
+      assert_equal({'hits' => (indx + 1).to_s, 'transactions' => (indx + 1).to_s}, RouterWrapper.config[:redis_count].hgetall(keys.first))
```
Comment (fab-girard) [inline test/api/v01/isoline_test.rb:76]: I don't understand how this test can be OK in Count api calls commit...

File: test/test_helper.rb [hunk 7]
```diff
@@ -35,3 +38,9 @@
 
 require 'minitest/autorun'
 require 'rack/test'
+
+module FakeRedis
+  def teardown
+    RouterWrapper.config[:redis_count].flushall
+  end
+end
```
Comment (fab-girard) [inline test/test_helper.rb:46]: empty line missing at the end: your editor is not correctly configured!

File: Gemfile.lock [hunk 8]
```diff
@@ -1,6 +1,7 @@
 GEM
   remote: https://rubygems.org/
   specs:
+    actionpack (1.4.0)
```
Comment (fab-girard) [inline Gemfile.lock:4]: gem is really `1.4.0` at this stage?

File: api/v01/api_base.rb [hunk 9]
```diff
@@ -22,15 +22,19 @@ module Api
   module V01
     class APIBase < Grape::API
 
-      def self.services(api_key)
```
Comment (fab-girard) [inline api/v01/api_base.rb]: you should move this diff in access file commit

File: api/v01/api_base.rb [hunk 10]
```diff
@@ -22,15 +22,19 @@ module Api
...
+      def self.profile(api_key)
         raise 'Profile missing in configuration' unless ::RouterWrapper.config[:profiles].key? ::RouterWrapper.access[api_key][:profile]
 
-        ::RouterWrapper.config[:profiles][::RouterWrapper.access[api_key][:profile]]
+        ::RouterWrapper.config[:profiles][::RouterWrapper.access[api_key][:profile]].deep_merge(
```
Comment (fab-girard) [inline api/v01/api_base.rb:28]: you should move this diff in access file commit

File: api/v01/isoline.rb [hunk 11]
```diff
@@ -83,9 +83,10 @@ class Isoline < APIBase
           ]
         }
         get do
+          count :isolines
           params[:speed_multiplier] = params[:speed_multiplicator] if !params[:speed_multiplier]
           params[:speed_multiplier_area] = params[:speed_multiplicator_area] if params[:speed_multiplicator_area] && params[:speed_multiplicator_area].size > 0 && (!params[:speed_multiplier_area] || params[:speed_multiplier_area].size == 0)
-          isoline params
+          isoline params, operation: :isolines
```
[the remark this answers is not in the document]
Comment (alain-andre) [reply api/v01/isoline.rb]: I remove it.

File: api/v01/isoline.rb [hunk 12]
```diff
@@ -118,6 +120,7 @@ def isoline(params)
 
           results = RouterWrapper::wrapper_isoline(APIBase.services(params[:api_key]), params)
           results[:router][:version] = 'draft'
+          count_incr :operation, transactions: APIBase.count_locations(params['loc'])
```
[the remark this answers is not in the document]
Comment (alain-andre) [reply api/v01/isoline.rb]: You are right, it was fixed in next commit, I fix it in this one.

File: api/v01/matrix.rb (hunk 1 above)
Comment (alain-andre) [reply api/v01/matrix.rb]: removed.

File: api/v01/matrix.rb (hunk 2 above)
Comment (alain-andre) [reply api/v01/matrix.rb]: done.

File: api/v01/route.rb (hunk 4 above)
Comment (alain-andre) [reply api/v01/route.rb]: done

File: api/v01/route.rb (hunk 3 above)
Comment (alain-andre) [reply api/v01/route.rb]: done.

File: api/v01/route.rb (hunk 5 above)
Comment (alain-andre) [reply api/v01/route.rb]: removed.

File: test/test_helper.rb (hunk 7 above)
Comment (alain-andre) [reply test/test_helper.rb]: done.

File: test/api/v01/isoline_test.rb (hunk 6 above)
Comment (alain-andre) [reply test/api/v01/isoline_test.rb:66]: I don't understand why it should not work. There is 2 requests each one has 1 transaction. so first time the count is 1 hit, 1 transaction and second time it's 2 hits, 2 transactions.

**Review by fab-girard — APPROVED** [review]

**Pull request merged**